mephisto/abstractions/providers/mturk/mturk_unit.py

```diff
@@ -109,11 +109,52 @@ def get_requester(self) -> "MTurkRequester":
             self.__requester = cast("MTurkRequester", super().get_requester())
         return self.__requester
 
+    def set_db_status(self, status: str) -> None:
+        """
+        Set the status reflected in the database for this Unit
+        """
+        super().set_db_status(status)
+        if status == AssignmentState.COMPLETED:
+            agent = self.get_assigned_agent()
+            if agent is not None:
+                agent_status = agent.get_status()
+                if agent_status == AgentState.STATUS_IN_TASK:
```

Comment:
Do we switch agent status from in_task to something else, regardless of reconcile failing or not?

Reply:
The reconciliation should put a final task act into the message queue for the agent being reconciled, which may be sufficient to move them to complete in the `TaskRunner`. This is definitely not a guarantee though.